src/strands/types/_events.py

```diff
@@ -373,4 +374,90 @@ def __init__(self, reason: str | Exception) -> None:
 
 class AgentResultEvent(TypedEvent):
     def __init__(self, result: "AgentResult"):
-        super().__init__({"result": result})
+        super().__init__({"agent_result": True, "result": result})
+
+
+class MultiAgentResultEvent(TypedEvent):
+    """Event emitted when multi-agent execution completes with final result."""
+
+    def __init__(self, result: "MultiAgentResult") -> None:
+        """Initialize with multi-agent result.
+
+        Args:
+            result: The final result from multi-agent execution (SwarmResult, GraphResult, etc.)
+        """
+        super().__init__({"multi_agent_result": True, "result": result})
+
+
+class MultiAgentNodeStartEvent(TypedEvent):
+    """Event emitted when a node begins execution in multi-agent context."""
+
+    def __init__(self, node_id: str, node_type: str) -> None:
+        """Initialize with node information.
+
+        Args:
+            node_id: Unique identifier for the node
+            node_type: Type of node ("agent", "swarm", "graph")
+        """
+        super().__init__({"multi_agent_node_start": True, "node_id": node_id, "node_type": node_type})
+
+
+class MultiAgentNodeStopEvent(TypedEvent):
+    """Event emitted when a node stops execution.
+
+    Similar to EventLoopStopEvent but for individual nodes in multi-agent orchestration.
+    Provides the complete NodeResult which contains execution details, metrics, and status.
+    """
+
+    def __init__(
+        self,
+        node_id: str,
+        node_result: "NodeResult",
+    ) -> None:
+        """Initialize with stop information.
+
+        Args:
+            node_id: Unique identifier for the node
+            node_result: Complete result from the node execution containing result,
+                execution_time, status, accumulated_usage, accumulated_metrics, and execution_count
+        """
+        super().__init__(
+            {
+                "multi_agent_node_stop": True,
+                "node_id": node_id,
+                "node_result": node_result,
+            }
+        )
+
+
+class MultiAgentHandoffEvent(TypedEvent):
+    """Event emitted during agent handoffs in Swarm."""
+
+    def __init__(self, from_node: str, to_node: str, message: str) -> None:
+        """Initialize with handoff information.
+
+        Args:
+            from_node: Node ID handing off control
+            to_node: Node ID receiving control
+            message: Handoff message explaining the transfer
+        """
+        super().__init__({"multi_agent_handoff": True, "from_node": from_node, "to_node": to_node, "message": message})
```

Comment:
`message` right now is the way to detect Agent messages - [our docs' example would fail if this new event was introduced and returned](https://strandsagents.com/latest/documentation/docs/user-guide/concepts/streaming/overview/#event-loop-lifecycle-example).